### ISSUE TITLE
login fails

### ISSUE THREAD

**Issue by Proulx-S** [zerolfx/copilot.el#65]
Hello,

I am having trouble login to GitHub. Disclaimer: emacs newbie here.

copilot-login gives the following
![image](https://user-images.githubusercontent.com/68520323/203658796-737f0cb6-5f56-41f1-bdcb-d4188623db06.png)

which brings me to this emacs browser
![image](https://user-images.githubusercontent.com/68520323/203658969-b917b07e-db6c-4724-b437-7a1aaa1a80e3.png)

That is where I get confused.
(1) What are the two fields after password? The first is hidden and the second not, so I put my password in the first. I don't know what to do with the one-time code from the previous step, so I put it in the second field--leaving it blank does not seem to change anything.
(2) How to click the "sign in" button? I can't seem to make it work.
(3) What are these multiple "submit" buttons?

In any case, just pressing Enter in the mini buffer brings me to this:
![image](https://user-images.githubusercontent.com/68520323/203659204-2dfee360-33c9-4702-a836-8fad579252ce.png)
and copilot-diagnose just reports that I am not logged in: `Copilot error: (:code 1000 :message "Not authenticated: NotSignedIn")`

Here is the full content of the \*Messages\* buffer
```
Loading /etc/emacs/site-start.d/00debian.el (source)...done
For information about GNU Emacs and the GNU system, type C-h C-a.
Copilot agent started.
Contacting host: github.com:443
uncompressing publicsuffix.txt.gz...done
Input field
Verifying...
Authentication failure: Timed out
Authenticated as GitHub user nil.
You can run the command ‘copilot-login’ with M-x c-logi RET
Authenticated as GitHub user nil.
[jsonrpc] Server exited with status 9
Copilot agent started.
Copilot error: (:code 1000 :message "Not authenticated: NotSignedIn")
```

and of the \*copilot events\* buffer (2 different attempts)
```
(:jsonrpc "2.0" :id 2 :result "OK")
[server-reply] (id:3) ERROR Wed Nov 23 17:56:04 2022:
(:jsonrpc "2.0" :id 3 :error
	  (:code 1000 :message "Not authenticated: NotSignedIn"))
[client-request] (id:4) Wed Nov 23 17:57:36 2022:
(:jsonrpc "2.0" :id 4 :method "getCompletions" :params
	  (:doc
	   (:source ";; This buffer is for text that is not saved, and for Lisp evaluation.\n;; To create a file, visit it with C-x C-f and enter text in its buffer.\n\n\n" :tabSize 8 :indentSize 8 :insertSpaces :json-false :path nil :uri "" :relativePath "" :languageId "lisp-interaction" :position
		    (:line 3 :character 0))))
[server-reply] (id:4) ERROR Wed Nov 23 17:57:36 2022:
(:jsonrpc "2.0" :id 4 :error
	  (:code 1000 :message "Not authenticated: NotSignedIn"))
[internal] (id:4) ERROR Wed Nov 23 17:57:36 2022:
(:message "error ignored, status set (Not authenticated: NotSignedIn)" :id 4 :error 1000)
```
```
(:jsonrpc "2.0" :method "LogMessage" :params
	  (:level 0 :message "[DEBUG] [agent] [2022-11-23T23:06:39.119Z] Telemetry initialized" :metadataStr "[DEBUG] [agent] [2022-11-23T23:06:39.119Z]" :extra
		  ["Telemetry initialized"]))
[server-reply] (id:2) Wed Nov 23 18:06:39 2022:
(:jsonrpc "2.0" :id 2 :result "OK")
[server-reply] (id:3) Wed Nov 23 18:06:39 2022:
(:jsonrpc "2.0" :id 3 :result
	  (:status "PromptUserDeviceFlow" :userCode "1406-2ED0" :verificationUri "https://github.com/login/device" :expiresIn 899 :interval 5))
[client-request] (id:4) Wed Nov 23 18:07:09 2022:
(:jsonrpc "2.0" :id 4 :method "signInConfirm" :params
	  (:userCode "1406-2ED0"))
[client-request] (id:5) Wed Nov 23 18:07:19 2022:
(:jsonrpc "2.0" :id 5 :method "checkStatus" :params
	  (:dummy "checkStatus"))
[server-reply] (id:5) Wed Nov 23 18:07:19 2022:
(:jsonrpc "2.0" :id 5 :result
	  (:status "NotSignedIn"))
```
And for the sake of completeness, here is my .emacs init file
```
;; Disable package.el in favor of straight.el
(setq package-enable-at-startup nil)

(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(custom-enabled-themes '(wheatgrass)))
(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 )

;; Install straight.el
(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
      (bootstrap-version 6))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/radian-software/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))
;; Install use-package
(straight-use-package 'use-package)
;; Configure use-package to use straight.el by default
(use-package straight
             :custom (straight-use-package-by-default t))

;; Load copilot
(use-package copilot
  :straight (:host github :repo "zerolfx/copilot.el" :files ("dist" "*.el"))
  :ensure t)
;; you can utilize :map :hook and :config to customize copilot
(add-hook 'prog-mode-hook 'copilot-mode)
;; Configure copilot
(defun my/copilot-tab ()
  (interactive)
  (or (copilot-accept-completion)
      (indent-for-tab-command)))
(with-eval-after-load 'copilot
  (define-key copilot-mode-map (kbd "<tab>") #'my/copilot-tab))
;; Configure nodejs dependency
(setq copilot-node-executable "/home/proulx-s/.nvm/versions/node/v16.18.1/bin/node")
```

I am running emacs 27.1 in Ubuntu 18.04 LTS through WSL2.

Thanks for the good work and for the help!
Sébastien

**Comment by zerolfx** [comment]
Please use a graphical browser instead to log in. (URL is the same.)

**Comment by Proulx-S** [comment]
Oh, thanks for the tip. I did not realize this was an actual webpage displayed within emacs.

So that URL into my browser brings me to this github page
![image](https://user-images.githubusercontent.com/68520323/203678980-77ac4bcf-460d-42b4-a317-b9dfce360286.png)
and entering the one-time code does produce this other page allowing to "authorize copilot plugin". But then what?

But back to emacs, nothing changed. Pressing "enter" in the minibuffer still gives the "Authenticated as Github user nil" message, and the \*messages\* and \*copilot events\* logs are the same.

**Comment by zerolfx** [comment]
Did you authorize the plugin after entering the one-time code?

**Comment by Proulx-S** [comment]
Entering the one-time code brought me to a second page with two buttons: (1) cancel, I think, and (2) authorize the plugin. I clicked (2).

**Comment by zerolfx** [comment]
Unfortunately, I can't reproduce this problem.

**Comment by Proulx-S** [comment]
Just started fresh after leaving this for a while. Working now. Not sure what did the trick--I reinstalled everything including emacs.

One thing for sure, for login in, I needed to manually go to https://github.com/login/device on my web browser, which was not clear from the interactive instructions.

What worked in summary:
(1) copilot-login in emacs
(2) follow the interactive instructions until it give you the one-time code
(3) ignore the github buffer that opens up and instead navigate to https://github.com/login/device in your web browser
(4) validate the one-time code there
(5) come back to the interactive instructions

Thanks for the help!
Sébastien
